connectors/sources/sharepoint_online.py

```diff
@@ -89,7 +98,7 @@ class NotFound(Exception):
 
 
 class InternalServerError(Exception):
-    """Internal exception class to handle 500s from the API, which could sometimes also mean NotFound."""
+    """Exception class to indicate that something went wrong on the server side."""
```

Comment:
It follows the practice with `NotFound` for example - "Internal" here means, that it's used for internal logic of the clients :)